drivers/spi/spim_nrf52.c

```diff
@@ -158,12 +158,15 @@ static int spim_nrf52_configure(struct device *dev,
 		return -EINVAL;
 	}
 
-	spim->CONFIG = (flags & SPI_TRANSFER_LSB) ? SPIM_CONFIG_ORDER_LsbFirst
-		       : SPIM_CONFIG_ORDER_MsbFirst;
-	spim->CONFIG |= (flags & SPI_MODE_CPOL) ? SPIM_CONFIG_CPOL_ActiveLow
-			: SPIM_CONFIG_CPOL_ActiveHigh;
-	spim->CONFIG |= (flags & SPI_MODE_CPHA) ? SPIM_CONFIG_CPHA_Trailing
-			: SPIM_CONFIG_CPHA_Leading;
+	spim->CONFIG = (flags & SPI_TRANSFER_LSB) ?
+		       SPIM_CONFIG_ORDER_LsbFirst :
+		       SPIM_CONFIG_ORDER_MsbFirst;
+	spim->CONFIG |= (flags & SPI_MODE_CPOL) ?
+			(SPIM_CONFIG_CPOL_ActiveLow << SPIM_CONFIG_CPOL_Pos) :
+			(SPIM_CONFIG_CPOL_ActiveHigh << SPIM_CONFIG_CPOL_Pos);
+	spim->CONFIG |= (flags & SPI_MODE_CPHA) ?
```

Comment:
It doesn't depend on the CPOL because CPHA only define if it's the trailing or leading edge. However I did a mistake, sorry, if CPHA = 1 the data must be sampled on the trailing edge.

	spim->CONFIG |= (flags & SPI_MODE_CPHA) ?
			(SPIM_CONFIG_CPHA_Trailing << SPIM_CONFIG_CPHA_Pos):
			(SPIM_CONFIG_CPHA_Leading << SPIM_CONFIG_CPHA_Pos) ;